### PR TITLE
[Jormungandr] Manage equipment providers

### PR DESCRIPTION
**Pull request by mbounabi**
- [x] Merge navitia-proto: https://github.com/CanalTP/navitia-proto/pull/121

### CODE REVIEW

File: source/jormungandr/jormungandr/instance.py [hunk 1]
```diff
@@ -668,3 +671,31 @@ def get_ridesharing_journeys_with_feed_publishers(self, from_coord, to_coord, pe
             res.extend(rsjs)
             fps.add(fp)
         return res, fps
+
+    def manage_equipments(self, response):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/instance.py]: I would prefer to have this logic somewhere else, probably in the equipments manager, or something else.
`Instance` is already quite big and do way to many thing

File: source/jormungandr/jormungandr/instance.py [hunk 2]
```diff
@@ -668,3 +671,31 @@ def get_ridesharing_journeys_with_feed_publishers(self, from_coord, to_coord, pe
...
+        def get_from_to_stop_points_of_journeys(journeys):
+            """
+            :param journeys: the pb journey response
+            :return: generator of stop points that can be 'origin' or 'destination'
+            """
+            places = itertools.chain(*[(s.origin, s.destination) for j in journeys for s in j.sections])
+            return (
+                place.stop_point
+                for place in places
+                if place.embedded_type == type_pb2.NavitiaType.Value('STOP_POINT')
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/instance.py]: you could also check that `place.stop_point` is set/initialized in proto

File: source/jormungandr/jormungandr/instance.py (hunk 1 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/instance.py]: :heavy_check_mark:

File: source/jormungandr/jormungandr/instance.py (hunk 2 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/instance.py]: :heavy_check_mark:

File: source/jormungandr/jormungandr/equipments/equipment_provider_manager.py [hunk 3]
```diff
@@ -41,12 +41,18 @@ def __init__(self, equipment_providers_configuration):
...
-            self._equipment_providers_legacy[configuration['key']] = self._init_class(
-                configuration['class'], arguments
-            )
+    def init_providers(self, providers_keys):
+        """
+        Create equipment providers only if defined in the Jormungandr instance and not already created
+        :param providers_keys: list of providers defined in the instance
+        """
+        for provider in self.providers_config:
+            if provider['key'].lower() in providers_keys and provider['key'].lower() not in dict(
```
Comment (nonifier) [inline source/jormungandr/jormungandr/equipments/equipment_provider_manager.py]: It's a shame to be doing `provider['key'].lower()` 3 times when once is necessary ;)

File: source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py [hunk 4]
```diff
@@ -346,6 +348,29 @@ class CommercialModeSerializer(PbGenericSerializer):
...
+
+
+class EffectSerializer(PbNestedSerializer):
+    label = base.PbStrField(display_none=True, required=False)
+
+
+class CurrentAvailabilitySerializer(PbNestedSerializer):
+    status = EnumField(pb_type=CurrentAvailability.EquipmentStatus, display_none=False, required=False)
+    periods = PeriodSerializer(many=True, display_none=False, required=False)
+    updated_at = base.PbStrField(display_none=True, required=False)
```
Comment (nonifier) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py]: It's a bit weird to display `update_at` when none (especially that it's the only `display_none=True` from the object).

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/jormungandr/equipments/equipment_provider_manager.py (hunk 3 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/equipments/equipment_provider_manager.py]: image

File: source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py (hunk 4 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py]: Obviously, Copy-Paste gone wrong!!

**Review by nonifier — APPROVED** [review]
:+1:

File: source/jormungandr/jormungandr/equipments/sytral.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        :return: data received from the webservice
+
+        NOTE: CALL TO ACTUAL WEBSERVICE WILL BE DONE WHEN IT WILL BE AVAILABLE (NAVP-1231)
+        """
+        logging.getLogger(__name__).debug('sytralRT RT service , call url : {}'.format(self.url))
+        pass
+
+    def _process_data(self, data, stop_points_list):
+        """
+        For each stop point in te journeys response, the structure 'equipment_details' is updated if the corresponding code is present
```
Comment (benoit-bst) [inline source/jormungandr/jormungandr/equipments/sytral.py]: For each stop point `within` journeys response

File: source/jormungandr/jormungandr/equipments/sytral.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        :param data: equipments data received from the webservice
+        :param stop_points_list: list of stop_points from the protobuf response
+        """
+        for st in stop_points_list:
+            for code in st.codes:
+                if "TCL_" in code.type:
```
Comment (benoit-bst) [inline source/jormungandr/jormungandr/equipments/sytral.py]: Just a question, is it better to have it in parameter ? I don't know if it's smarty to hard coded it

**Review by benoit-bst — APPROVED** [review]

**Pull request merged**